packages/electron/lib/electron.ts

```
@@ -0,0 +1,194 @@
+
+  return _install.packageAndExit(...args)
+}
+
+export function getElectronVersion () {
+  return _install.getElectronVersion()
+}
+
+/**
+ * Returns the Node version bundled inside Electron.
+ */
+export function getElectronNodeVersion () {
+  debugElectron('getting Electron Node version')
+
+  const args = []
+
+  if (isSandboxNeeded()) {
+    args.push('--no-sandbox')
+  }
+
+  // runs locally installed "electron" bin alias
+  const localScript = path.join(__dirname, 'print-node-version.js')
+
+  debugElectron('local script that prints Node version %s', localScript)
+
+  args.push(localScript)
+
+  const options = {
+    preferLocal: true, // finds the "node_modules/.bin/electron"
+    timeout: 10000, // prevents hanging Electron if there is an error for some reason
+  }
+
+  debugElectron('Running Electron with %o %o', args, options)
+
+  return execa('electron', args, options)
+  .then((result) => result.stdout)
+}
+
+export function icons () {
+  return _install.icons()
+}
+
+export function cli (argv = []) {
+  const opts = minimist(argv)
+
+  debugElectron('cli options %j', opts)
+
+  if (opts.install) {
+    return installIfNeeded()
+  }
+
+  if (opts.help || opts.h) {
+    console.log(`
+Usage: cypress-electron [options] [app-path]
+
+Options:
+  --install    Install/build the Electron binary
+  --help, -h   Show this help message
+
+Examples:
+  cypress-electron --install
+  cypress-electron /path/to/your/app
+`)
+
+    return
+  }
+
+  const pathToApp = argv[0]
+
+  if (pathToApp) {
+    return open(pathToApp, argv)
+  }
+
+  throw new Error('No path to your app was provided.')
+}
+
+export async function open (appPath: string, argv: string[], cb?: (code: number) => void) {
+  debugElectron('opening %s', appPath)
+
+  appPath = path.resolve(appPath)
+  const dest = paths.getPathToResources('app')
+
+  debugElectron('appPath %s', appPath)
+
+  debugElectron('dest path %s', dest)
+
+  try {
+    await fs.stat(appPath)
```

previously this was `access`. I am guessing we mostly just care for existence and this is fine and actually more appropriate

---

no, access is the right thing - the debug had me mixed up (and that fs-extra's remove fails silently if the fd does not exist).